src/NoFrixion.MoneyMoov/Models/Roles/RoleEvent.cs

```diff
@@ -0,0 +1,89 @@
+﻿// -----------------------------------------------------------------------------
+//  Filename: RoleEvent.cs
+// 
+//  Description: A model that represents an event produced by a MoneyMoov
+// Role execution.
+// 
+//  Author(s):
+//  Axel Granillo (axel@nofrixion.com)
+// 
+//  History:
+//  18 07 2025  Axel Granillo   Created, Remote, Mexico City, Mexico.
+// 
+//  License:
+//  MIT.
+// -----------------------------------------------------------------------------
+
+using NoFrixion.Common.Permissions;
+using NoFrixion.MoneyMoov.Enums;
+
+namespace NoFrixion.MoneyMoov.Models;
+
+/// <summary>
+/// A model that represents an event produced by a MoneyMoov Role execution.
+/// </summary>
+public class RoleEvent
+{
+    public Guid ID { get; set; }
+    
+    public Guid RoleID { get; set; }
+    
+    public string? Notes { get; set; }
+    
+    public RoleEventTypeEnum Type { get; set; }
+    
+    public DateTimeOffset Inserted { get; set; }
+    
+    public string? AuthoriserHash { get; set; }
+
+    /// <summary>
+    /// The name the role had at the time of the event.
+    /// </summary>
+    public string? Name { get; set; }
+
+    /// <summary>
+    /// The description the role had at the time of the event.
+    /// </summary>
+    public string? Description { get; set; }
+
+    /// <summary>
+    /// The merchant permissions the role had at the time of the event.
+    /// </summary>
+    public MerchantPermissions MerchantPermissions { get; set; }
+
+    /// <summary>
+    /// The account permissions the role had at the time of the event.
+    /// </summary>
+    public AccountPermissions AccountPermissions { get; set; }
+    
+    /// <summary>
+    /// The client session timeout seconds the role had at the time of the event.
+    /// </summary>
+    public int? ClientSessionTimeoutSeconds { get; set; }
+
+    /// <summary>
+    /// If the role was a system role at the time of the event.
+    /// </summary>
+    public bool? IsSystem { get; set; }
+    
+    /// <summary>
+    /// For <see cref="RoleEventTypeEnum.AssignedToUser"/> and <see cref="RoleEventTypeEnum.RemovedFromUser"/> event types.
+    /// The user that was assigned to or removed from the role at the time of the event.
+    /// </summary>
+    public Guid? AssignationEventUserID { get; set; }
```

Comment:
The original plan was to have role, user role assigments and user account assignments as 3 separate event tables rather than trying to do it all in a single event table. Can we have a chat about it on Monday to dig into it before going down this single event table path?